torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -191,6 +191,12 @@ def _init_mp_dtypes(self) -> None:
 
     def lazy_init(self):
         # Lazy init should be idempotent
+        # User may initialize params again after meta init.
+        # For example, we initialize DoRA magnitudes base on
+        # other parameters loaded from the state dict.
+        with torch.no_grad():
+            for fsdp_param in self.fsdp_params:
+                fsdp_param.reset_sharded_param()
```

Comment:
I think to fix https://github.com/pytorch/pytorch/issues/132978
```suggestion
        if self.is_sharded:
            with torch.no_grad():
                for fsdp_param in self.fsdp_params:
                    fsdp_param.reset_sharded_param()
```

Reply:
just tested and the gap is still in DDP(FSDP).  FSDPModule params are "unsharded" to `torch.Tensor` already before calling `.unshard`, but `FSDPParamGroup.is_sharded` is still `True` 

https://github.com/pytorch/pytorch/blob/a9954d22f89677fc22fb9764f1edb0818538478f/test/distributed/_composable/fsdp/test_fully_shard_comm.py#L1066